aci/resource_aci_cloudlb.go

```diff
@@ -0,0 +1,570 @@
+	d.Set("max_instance_count", models.StripQuotes(cloudLBAttrs.S("maxInstanceCount").String()))
+	d.Set("min_instance_count", models.StripQuotes(cloudLBAttrs.S("minInstanceCount").String()))
+	d.Set("mode", models.StripQuotes(cloudLBAttrs.S("mode").String()))
+	d.Set("name", models.StripQuotes(cloudLBAttrs.S("name").String()))
+	d.Set("native_lb_name", models.StripQuotes(cloudLBAttrs.S("nativeLBName").String()))
+	d.Set("package_model", models.StripQuotes(cloudLBAttrs.S("packageModel").String()))
+	d.Set("prom_mode", models.StripQuotes(cloudLBAttrs.S("promMode").String()))
+	d.Set("scheme", models.StripQuotes(cloudLBAttrs.S("scheme").String()))
+	d.Set("size", models.StripQuotes(cloudLBAttrs.S("size").String()))
+	d.Set("sku", models.StripQuotes(cloudLBAttrs.S("sku").String()))
+	d.Set("service_type", models.StripQuotes(cloudLBAttrs.S("svcType").String()))
+	d.Set("target_mode", models.StripQuotes(cloudLBAttrs.S("targetMode").String()))
+	d.Set("trunking", models.StripQuotes(cloudLBAttrs.S("trunking").String()))
+	d.Set("cloud_l4l7_load_balancer_type", models.StripQuotes(cloudLBAttrs.S("type").String()))
+
+	cloudLBChildCount, err := cloudLBCont.S("imdata").Index(0).S(CloudLBClassName).ArrayCount("children")
+	if err != nil {
+		return nil, err
+	}
+
+	cloudLBChildSubnetList := make([]string, 0)
+	cloudLBChildAaaDomainList := make([]string, 0)
+
+	cloudLBChild := cloudLBCont.S("imdata").Index(0).S(CloudLBClassName).S("children")
+	for i := 0; i < cloudLBChildCount; i++ {
+		cloudSubnetTDn := models.StripQuotes(cloudLBChild.Index(i).S(CloudRsLDevToCloudSubnetClassName).S("attributes").S("tDn").String())
+		if cloudSubnetTDn != "{}" {
+			cloudLBChildSubnetList = append(cloudLBChildSubnetList, cloudSubnetTDn)
+			continue
+		}
+		aaaDomainName := models.StripQuotes(cloudLBChild.Index(i).S("aaaRbacAnnotation").S("attributes").S("domain").String())
+		if aaaDomainName != "{}" {
+			cloudLBChildAaaDomainList = append(cloudLBChildAaaDomainList, fmt.Sprintf("uni/userext/domain-%s", aaaDomainName))
+			continue
+		}
+	}
+	d.Set("relation_cloud_rs_ldev_to_cloud_subnet", cloudLBChildSubnetList)
+	d.Set("aaa_domain_dn", cloudLBChildAaaDomainList)
+
+	d.SetId(cloudLBDistinguishedName)
+	return d, nil
+}
+
+func resourceAciCloudL4L7LoadBalancerImport(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	log.Printf("[DEBUG] %s: Beginning Import", d.Id())
+	aciClient := m.(*client.Client)
+	dn := d.Id()
+
+	schemaFilled, err := getAndSetRemoteCloudL4L7LoadBalancerAttributes(aciClient, dn, d)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Printf("[DEBUG] %s: Import finished successfully", d.Id())
+	return []*schema.ResourceData{schemaFilled}, nil
+}
+
+func resourceAciCloudL4L7LoadBalancerCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	log.Printf("[DEBUG] Cloud L4-L7 Load Balancer: Beginning Creation")
+	aciClient := m.(*client.Client)
+
+	annotation := d.Get("annotation").(string)
```

Comment:
annotation get can be done in map logic, since d is already passed into function

Reply:
Discussed offline.